30-remote-site-setup.sh

```diff
@@ -153,7 +158,18 @@ fi
 echo "Using Bosco tarball: $(bosco_findplatform --url)"
 for ruser in $users; do
     setup_ssh_config
-    [[ $cvmfs_wn_client == 'no' ]] && setup_endpoints_ini
+done
+
+###################
+# REMOTE COMMANDS #
+###################
+
+# We have to pick a user for SSH, may as well be the first one
+remote_os_info=$(fetch_remote_os_info "$(echo $users | head -n1)" "$remote_fqdn")
```

Comment:
```suggestion
remote_os_info=$(fetch_remote_os_info "$(printf "%s\n" $users | head -n1)" "$remote_fqdn")
```
assuming you're trying to get the first word in a space-separated list.

Reply:
Nope, `$users` is already separated by newlines:

```
blin@blin-latitude:~$ cat foo
"/osg/Role=NULL/Capability=NULL" osg01
"/GLOW/Role=htpc/Capability=NULL" osg02
blin@blin-latitude:~$ users=$(cat foo | awk '/^"[^"]+" +[a-zA-Z0-9\-\._]+$/ {print $NF}' | sort -u)
blin@blin-latitude:~$ echo "$users" | head -n1
osg01
``

Reply:
OK but right here you're not quoting `$users` which means the echo will display all of them on one line.

If you quote `$users`, you'll get leading and trailing spaces (I assume you want to get rid of that). If you don't quote `$users` then you'll get them space separated instead of newline separated and `head -n1` won't work.

Using unquoted users with `printf` will do the job:
```
$ users=$'   osg01\n   osg02   osg03    '
$ printf "%s\n" $users
osg01
osg02
osg03
```